internal/core/src/storage/Event.h

```diff
@@ -0,0 +1,162 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <map>
+#include <string>
+#include <memory>
+#include <vector>
+
+#include "common/Types.h"
+#include "storage/Types.h"
+#include "storage/PayloadStream.h"
+#include "storage/FieldData.h"
+
+namespace milvus::storage {
+
+struct EventHeader {
+    milvus::Timestamp timestamp_;
+    EventType event_type_;
+    int32_t event_length_;
+    int32_t next_position_;
+
+    EventHeader() {
+    }
+    explicit EventHeader(PayloadInputStream* input);
+
+    std::vector<uint8_t>
+    Serialize();
+};
+
+struct DescriptorEventDataFixPart {
+    int64_t collection_id;
+    int64_t partition_id;
+    int64_t segment_id;
+    int64_t field_id;
+    Timestamp start_timestamp;
+    Timestamp end_timestamp;
+    milvus::proto::schema::DataType data_type;
+
+    DescriptorEventDataFixPart() {
+    }
+    explicit DescriptorEventDataFixPart(PayloadInputStream* input);
+
+    std::vector<uint8_t>
+    Serialize();
+};
+
+struct DescriptorEventData {
+    DescriptorEventDataFixPart fix_part;
+    int32_t extra_length;
+    std::vector<uint8_t> extra_bytes;
+    std::map<std::string, std::string> extras;
+    std::vector<uint8_t> post_header_lengths;
+
+    DescriptorEventData() {
+    }
+    explicit DescriptorEventData(PayloadInputStream* input);
+
+    std::vector<uint8_t>
+    Serialize();
+};
+
+struct InsertEventData {
```

Comment:
"InsertEventData" and "IndexEventData" are redundant structures.
Only define "EventData" is fine, add "EventType" in it.